src/Repository/DonationRepository.php

```diff
@@ -26,4 +33,13 @@ public function findByEmailAddressOrderedByDonatedAt(string $email, string $orde
             ['donatedAt' => $order]
         );
     }
+
+    public function findSubscriptions(array $criteria): array
+    {
+        if (PayboxPaymentSubscription::NONE === $criteria['duration']) {
```

Comment:
I don't really understand this wrapper around `findBy` method. Could you explain why please?

I think this logic belongs to the validator and that's all.